DESIGN.md

```diff
@@ -0,0 +1,818 @@
+whether it may be worth further looking into it for signs of censorship.
+
+- `is_done` indicates whether this measurement is complete.
+
+- `is_failed` and `failure_msg` indicate whether and how the measurement failed.
+
+- `is_uploaded`, `is_upload_failed`, and `upload_failure_msg` indicate whether
+the measurement is uploaded, and whether and how the upload failed.
+
+- `test_keys` contains some interesting test keys that we may want to show
+to the user without pulling the full measurement JSON. This is an experiment
+specific serialized JSON. See below for what to expect for each experiment.
+
+- `test_name` is the name of the specific experiment (OONI experiments, aka
+nettests, are specified at https://github.com/ooni/spec).
+
+- `url`, `url_category_code`, and `url_country_code` all describe the URL
+used by this measurement, if applicable.
+
+The summary message instead is like:
+
+```JSON
+{
+  "fields": {
+    "total_count": 3,
+    "anomaly_count": 0,
+
+    "type": "measurement_summary",
+
+    "asn": 30722,
+    "data_usage_down": 15.4765625,
+    "data_usage_up": 4.556640625,
+    "network_country_code": "IT",
+    "network_name": "Vodafone Italia S.p.A.",
+    "start_time": "2019-10-03T07:59:10.325979Z",
+    "total_runtime": 3.141369
+  },
+  "level": "info",
+  "timestamp": "2019-10-03T16:42:00.22201+02:00",
+  "message": "measurement summary"
+}
+```
+
+where:
+
+- `anomaly_count` and `total_count` are the the total number of measurements
+emitted and the number of these that were anomalies.
+
+- `type` identifies this as a measurement summary message.
+
+- all other fields have already-describes semantics, which in this case of
+course is applied to the set of measurements as a whole.
+
+The `id` field of a `<measurement-entry>` message is key to get the
+raw measurement JSON, using `ooniprobe --batch show <id>`.
+
+#### Content of the test_keys field
+
+For the `"websites"` experiment group:
+
+```JSON
+{
+  "accessible": true,
+  "blocking": "none"
+}
+```
+
+where `accessible` indicates whether the URL is accessible and `blocking`
+indicates they way in which we believe the URL to be blocked.
+
+For the `"dash"` experiment in the `"performance"` group:
+
+```JSON
+{
+  "connect_latency": 0.06374096870422363,
+  "median_bitrate": 36724, 
+  "min_playout_delay": 0
+} 
+```
+
+where `"connect_latency"` is the time to establish a connection,
+`"median_bitrate"` is the median bitrate in kbit/s, and
+`"min_playout_delay`" is the minimum delay required to perform
+a smooth streaming, given the streaming speed.
+
+For the `"ndt"` experiment in the `"performance"` group:
+
+```JSON
+{
+  "upload": 21125.226149733466,
+  "download": 76153.12071218148,
+  "ping": 64
+}
+```
+
+where `"upload"` is the upload speed in kbit/s, `"download"` is the
+download speed in kbit/s, and `"ping"` is the minimum RTT in ms.
+
+For the experiments in the `"middlebox"` group:
+
+```JSON
+{}
```

Comment:
Is this unintended?

Reply:
I believe that since we only have to display a boolean of "middlebox detected" or "no middlebox detected" we just need the `is_anomaly` field